### PR TITLE
Quest Thandol Span (unfinished)

### PR DESCRIPTION
**Pull request by TracyTheBoy**
The unit it for the destroyed bridge still needs to be added 
Also need testing

### CODE REVIEW

File: src/WarcraftLegacies.Source/Quests/QuestThandolSpan.cs
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/QuestThandolSpan.cs:54]: You can shorthand this with an extension method. Something like `LegendNeutral.ThandolSpanDestroyed.Unit.SetLifePercent(0.05f)`

File: src/WarcraftLegacies.Source/Setup/Legends/LegendNeutral.cs
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Setup/Legends/LegendNeutral.cs:29]: I wonder if there is a good reason to use a Capital for this? You only need the destructable aspect right, so it could just be a plain destructable?

**Review by YakaryBovine — CHANGES_REQUESTED** [review]
review speedrun

File: src/WarcraftLegacies.Source/Quests/QuestThandolSpan.cs
Comment (TracyTheBoy) [reply src/WarcraftLegacies.Source/Quests/QuestThandolSpan.cs:26]: oh yea I just copy pasted that from the other quest. I figured I'd leave that to Sebas

File: src/WarcraftLegacies.Source/Quests/QuestThandolSpan.cs
Comment (YakaryBovine) [reply src/WarcraftLegacies.Source/Quests/QuestThandolSpan.cs:20]: Right, sorry, it *is* readonly because it's a get-only private property. But I think it's similar if it's just a readonly field instead.

File: src/WarcraftLegacies.Source/Setup/Legends/LegendNeutral.cs
Comment (TracyTheBoy) [reply src/WarcraftLegacies.Source/Setup/Legends/LegendNeutral.cs:29]: I need a unit to be able to use objectives. Or can I convert that somehow?

Comment (YakaryBovine) [reply src/WarcraftLegacies.Source/Setup/Legends/LegendNeutral.cs:29]: I think you should be able to just use the `ObjectiveControlUnit` objective instead, and use the `PreplacedUnitSystem` to pass in the appropriate unit as a parameter. You would then need to make the unit capturable separately, which you can do with the `MakeCapturable` extension method.

Comment (YakaryBovine) [reply src/WarcraftLegacies.Source/Setup/Legends/LegendNeutral.cs:29]: Sorry again, I failed to read properly. I see that the objective you're using is actually `ObjectiveLegendReachesFullHealth`. I think we should make a new `ObjectiveUnitReachesFullHealth`, but that could be done after this PR is merged if you'd prefer.

Comment (TracyTheBoy) [reply src/WarcraftLegacies.Source/Setup/Legends/LegendNeutral.cs:29]: Yea my bad, I don't need a capital anymore. I used a different objective at first which required a capital but not anymore so I'll changed that

**Review by YakaryBovine — APPROVED** [review]
excellent

**Pull request merged**
